src/application-insights/azext_applicationinsights/_params.py

```diff
@@ -0,0 +1,14 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+
+
+def load_arguments(self, _):
+    with self.argument_context('monitor app-insights query') as c:
+        c.argument('application', options_list=['--app-id'], help='GUID of the Application Insights applications')
```

Comment:
Are app-insights applications ARM resources? Asking to see if a user can refer to them through a ARM id.

Reply:
Yes, it accepts both. Should I accept the fully qualified ARM id here? Or accept the IDs parameter on each command and exclusively accept GUID here?

I also realized the default generation yields `app-id` as parameter name which goes against CLI guidelines; I can change this to `app` for all commands and accept ARM resources.

Reply:
Yes, please change to `app` or `application`. Also can you show me what the arm resource path/id for an application would like? 
I think ideally, this parameter should accept both the guid or the resource name (which would be similar to most other commands in the CLI).

Reply:
The GUID is accessible via an "API Access" blade from App Insights resource in Azure Portal. I'll include this info. 

Example resource: `/subscriptions/b98981de-4152-480f-a515-59b099299283/resourceGroups/ace-test/providers/microsoft.insights/components/ace-test`

Reply:
Yes, please include the information in the help and allow users to specify the application through resource-name/resource-group, etc. 
Take a look at: https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-the-ids-parameter